flake.nix

```diff
@@ -1,42 +1,189 @@
+        };
+        nixpkgs = pkgs;
+        # ... and construct a flake from the cabal.project file.
+        # We use cabalProject' to ensure we don't build the plan for
+        # all systems.
+        flake = (pkgs.haskell-nix.cabalProject' rec {
+          src = ./.;
+          name = "cardano-base";
+          compiler-nix-name = "ghc927";
+
+          # CHaP input map, so we can find CHaP packages (needs to be more
+          # recent than the index-state we set!). Can be updated with
+          #
+          #  nix flake lock --update-input CHaP
+          #
+          inputMap = {
+            "https://input-output-hk.github.io/cardano-haskell-packages" = inputs.CHaP;
+          };
+
+          # tools we want in our shell
+          shell.tools = {
+            cabal = "3.10.1.0";
+            ghcid = "0.8.8";
+            haskell-language-server = "latest";
+            hlint = {};
+            weeder = "2.4.1";
+          };
+          # Not we use pkgsBuildBuild, to make sure that even in the cross
+          # compilation setting, we don't run into issues where we pick tools
+          # for the target.
+          shell.buildInputs = with pkgs.pkgsBuildBuild; [
+            gitAndTools.git
+            sqlite-interactive
+          ];
+          shell.withHoogle = true;
+
+          # package customizations as needed. Where cabal.project is not
+          # specific enough, or doesn't allow setting these.
+          modules = [
+            ({pkgs, ...}: {
+              # Packages we wish to ignore version bounds of.
+              # This is similar to jailbreakCabal, however it
+              # does not require any messing with cabal files.
+              packages.katip.doExactConfig = true;
+
+              # split data output for ekg to reduce closure size
+              packages.ekg.components.library.enableSeparateDataOutput = true;
+              packages.cardano-binary.configureFlags = [ "--ghc-option=-Werror" ];
+              packages.cardano-crypto-class.configureFlags = [ "--ghc-option=-Werror" ];
+              # We need to override the pkgconfig libraries so that we can provide our fork of
+              # libsodium instead of the one from nixpkgs, which is what haskell.nix would
+              # otherwise choose. Unfortunately, this means that we also override any other
+              # pkgconfig libraries that haskell.nix would pick for us. So we also need to
+              # manually include those here.
+              packages.cardano-crypto-class.components.library.pkgconfig = pkgs.lib.mkForce [[ pkgs.libsodium-vrf pkgs.secp256k1 pkgs.blst ]];
+              packages.cardano-crypto-praos.components.library.pkgconfig = pkgs.lib.mkForce [[ pkgs.libsodium-vrf ]];
+              packages.slotting.configureFlags = [ "--ghc-option=-Werror" ];
+              enableLibraryProfiling = profiling;
+            })
+            ({pkgs, ...}: with pkgs; nixpkgs.lib.mkIf stdenv.hostPlatform.isWindows {
```

Comment:
```suggestion
            ({pkgs, ...}: with pkgs; lib.mkIf stdenv.hostPlatform.isWindows {
```

Reply:
this one can't come from pkgs here. as that leads to infinite recursion.